File: service/src/main/java/com/ricardococati/service/converter/MediaMovelSimplesConverter.java
package com.ricardococati.service.converter;

import com.ricardococati.model.entities.CandlestickDiario;
import com.ricardococati.model.entities.CandlestickSemanal;
import com.ricardococati.model.entities.MediaMovelSimples;
import com.ricardococati.model.entities.MediaMovelSimplesDiario;
import com.ricardococati.model.entities.MediaMovelSimplesSemanal;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Component;

@Slf4j
@Component
public class MediaMovelSimplesConverter {

  public MediaMovelSimplesDiario converterCandlestickDiarioToMediaMovelSimples(
      CandlestickDiario candlestickDiario) {
    return MediaMovelSimplesDiario
        .builder()
        .dtpreg(candlestickDiario.getDtpreg())
        .mediaMovelSimples(
            MediaMovelSimples
                .builder()
                .codneg(candlestickDiario.getCandlestick().getCodneg())
                .build())
        .build();
  }

  public MediaMovelSimplesSemanal converterCandlestickSemanalToMediaMovelSimples(
      CandlestickSemanal candlestickSemanal) {
    return MediaMovelSimplesSemanal
        .builder()
        .dtpregini(candlestickSemanal.getDtpregini())
        .dtpregfim(candlestickSemanal.getDtpregfim())
        .mediaMovelSimples(
            MediaMovelSimples
                .builder()
                .codneg(candlestickSemanal.getCandlestick().getCodneg())
                .build())
        .build();
  }

}

File: service/src/main/java/com/ricardococati/service/converter/MediaMovelSimplesConverter.kt
package com.ricardococati.service.converter

import com.ricardococati.model.entities.CandlestickDiario
import com.ricardococati.model.entities.CandlestickSemanal
import com.ricardococati.model.entities.MediaMovelSimples
import com.ricardococati.model.entities.MediaMovelSimplesDiario
import com.ricardococati.model.entities.MediaMovelSimplesSemanal
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Component

@Component
class MediaMovelSimplesConverter {

    fun converterCandlestickDiarioToMediaMovelSimples(
            candlestickDiario: CandlestickDiario): MediaMovelSimplesDiario {
        return MediaMovelSimplesDiario
                .builder()
                .dtpreg(candlestickDiario.dtpreg)
                .mediaMovelSimples(
                        MediaMovelSimples
                                .builder()
                                .codneg(candlestickDiario.candlestick!!.codneg)
                                .build())
                .build()
    }

    fun converterCandlestickSemanalToMediaMovelSimples(
            candlestickSemanal: CandlestickSemanal): MediaMovelSimplesSemanal {
        return MediaMovelSimplesSemanal
                .builder()
                .dtpregini(candlestickSemanal.dtpregini)
                .dtpregfim(candlestickSemanal.dtpregfim)
                .mediaMovelSimples(
                        MediaMovelSimples
                                .builder()
                                .codneg(candlestickSemanal.candlestick!!.codneg)
                                .build())
                .build()
    }

    companion object {
        private val log = LoggerFactory
                .getLogger(MediaMovelSimplesConverter::class.java)
    }

}